### PR TITLE
For #24210: Remove wrapper for metrics in the "events" category

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/metrics/MetricController.kt [hunk 1]
```diff
@@ -114,6 +115,10 @@ internal class ReleaseMetricController(
         Component.FEATURE_PROMPTS to LoginDialogFacts.Items.SAVE -> {
             LoginDialog.saved.record(NoExtras())
         }
+
+        Component.BROWSER_TOOLBAR to ToolbarFacts.Items.MENU -> {
+            Events.toolbarMenuVisible.record(mozilla.telemetry.glean.private.NoExtras())
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/components/metrics/MetricController.kt:120]: nit: `import mozilla.telemetry.glean.private.NoExtras`

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt [hunk 2]
```diff
@@ -592,20 +593,19 @@ class DefaultSessionControlController(
...
             val searchAccessPoint = Event.PerformedSearch.SearchAccessPoint.ACTION
-            searchAccessPoint.let { sap ->
+            val event = searchAccessPoint.let { sap ->
                 MetricsUtils.createSearchEvent(
                     searchEngine,
                     store,
                     sap
                 )
             }
+            event?.let { activity.metrics.track(it) }
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt:607]: Should we still be calling metrics.track here?

File: app/src/main/java/org/mozilla/fenix/settings/TabsSettingsFragment.kt [hunk 3]
```diff
@@ -104,12 +103,10 @@ class TabsSettingsFragment : PreferenceFragmentCompat() {
     }
 
     private fun sendTabViewTelemetry() {
-        val metrics = requireContext().components.analytics.metrics
-
         if (listRadioButton.isChecked && !gridRadioButton.isChecked) {
-            metrics.track(TabViewSettingChanged(Type.LIST))
+            Events.tabViewChanged.record(Events.TabViewChangedExtra("list"))
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/settings/TabsSettingsFragment.kt:107]: Looking at the original code we seems to be reporting `type.toString()` which makes it "LIST" and "GRID" (all cap).  Can you please confirm?

**Review by rocketsroger — COMMENTED** [review]
Looks great! And with tests!
Got some nits and clarification questions.

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt [hunk 4]
```diff
@@ -121,17 +122,27 @@ class SettingsFragment : PreferenceFragmentCompat() {
...
                 try {
-                    context?.let { context ->
-                        context.components.analytics.metrics.track(
-                            Event.PreferenceToggled(
-                                key,
-                                sharedPreferences.getBoolean(key, false),
-                                context
-                            )
-                        )
+                    val booleanPreferenceTelemetryAllowList = listOf(
```
[the remark this answers is not in the document]
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt]: Done

File: app/src/main/java/org/mozilla/fenix/settings/TabsSettingsFragment.kt (hunk 3 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/settings/TabsSettingsFragment.kt:107]: In the original `TabViewSettingChanged` a conversion was used:`type.toString().lowercase(Locale.ROOT)` so I tried to keep the original extra keys, even the letter case, although there was not a strict rule (uppercase or lowercase) enforced throughout the app.

File: app/src/test/java/org/mozilla/fenix/HomeActivityTest.kt [hunk 5]
```diff
@@ -50,10 +49,10 @@ class HomeActivityTest {
         val launcherIntent = Intent(Intent.ACTION_MAIN).apply {
             addCategory(Intent.CATEGORY_LAUNCHER)
         }.toSafeIntent()
-        assertEquals(Event.OpenedApp.Source.APP_ICON, activity.getIntentSource(launcherIntent))
+        assertEquals("APP_ICON", activity.getIntentSource(launcherIntent))
```
[the remark this answers is not in the document]
Comment (mcarare) [reply app/src/test/java/org/mozilla/fenix/HomeActivityTest.kt:52]: IMO the expected values in tests should be hardcoded to follow the description of the extra keys from the metric, ensuring the specifications from the "setting" file are followed by the code.

File: app/src/main/java/org/mozilla/fenix/settings/TabsSettingsFragment.kt (hunk 3 above)
Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/settings/TabsSettingsFragment.kt:107]: Ah i see, it is using lowercase.  Thanks for confirming.

**Review by rocketsroger — APPROVED** [review]
🚢

**Pull request merged**